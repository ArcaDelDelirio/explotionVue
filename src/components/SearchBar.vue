<script>
import DataGallery from './Gallery';
export default {
  name: 'SearchBar',
  components: {
    DataGallery,
  },
  data() {
    return {
      searchUser: '',
    };
  },
};
</script>


<template>
  <article class="container">
    <div class="main-title">
      <h1>Search on Database</h1>
    </div>
    <div class="search-container">
      <form class="search-form">
        <input
          class="search-input"
          type="text"
          placeholder="ex:Title"
          name="search"
          v-model="searchUser"
        />
        <button class="search-button">
          <i>
            <font-awesome-icon icon="search" />
          </i>
        </button>
      </form>
    </div>
    <div id="result">
      <DataGallery :searchQuery="searchUser" />
    </div>
  </article>
</template>

<style scoped>
.container {
  display: grid;
  grid-template-columns: 25% 50% 25%;
  grid-template-rows: 10% 5% 85%;
}
.main-title {
  grid-row: 1;
  grid-column: 2 / 3;
}
.search-container {
  display: flex;
  grid-column: 2 / 3;
  grid-row: 2;
}
.search-form {
  flex-grow: 1;
  display: flex;
}
.search-input {
  flex-grow: 2;
}
#result {
  grid-row: 3;
  grid-column: 2/3;
}
</style>
