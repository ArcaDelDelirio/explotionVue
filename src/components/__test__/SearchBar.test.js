import { shallowMount, createLocalVue } from '@vue/test-utils';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faNodeJs } from '@fortawesome/free-brands-svg-icons';
import {
  faServer,
  faSearch,
  faCircle,
} from '@fortawesome/free-solid-svg-icons';
import SearchBar from '../SearchBar';

describe('SearchBar View', () => {
  it('Render the component', () => {
    const localVue = createLocalVue();
    library.add(faNodeJs, faServer, faSearch, faCircle);
    localVue.component('font-awesome-icon', FontAwesomeIcon);
    const wrapper = shallowMount(SearchBar, { localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
