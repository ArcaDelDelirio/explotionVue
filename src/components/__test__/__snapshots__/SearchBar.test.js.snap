// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar View Render the component 1`] = `
<article class="container">
  <div class="main-title">
    <h1>Search on Database</h1>
  </div>
  <div class="search-container">
    <form class="search-form"><input type="text" placeholder="ex:Title" name="search" class="search-input"> <button class="search-button"><i>
          <font-awesome-icon-stub icon="search"></font-awesome-icon-stub>
        </i></button></form>
  </div>
  <div id="result">
    <datagallery-stub searchquery=""></datagallery-stub>
  </div>
</article>
`;
